test/Dialect/FIRRTL/legacy-wiring.mlir

```diff
@@ -206,3 +206,34 @@ firrtl.circuit "IntWidths" attributes {
     firrtl.connect %x, %invalid_ui1 : !firrtl.uint, !firrtl.uint
   }
 }
+
+// -----
+
+// Check direction of compatibility using const/non-const issue encountered (#6819).
+firrtl.circuit "Issue6819" attributes {
+ rawAnnotations = [
+  {
+    class = "firrtl.passes.wiring.SourceAnnotation",
+    target = "~Issue6819|Bar>y",
+    pin = "xyz"
+  },
+  {
+    class = "firrtl.passes.wiring.SinkAnnotation",
+    target = "~Issue6819|Issue6819>x",
+    pin = "xyz"
+  }
+  ]} {
+  firrtl.module private @Bar() {
+    %y = firrtl.wire interesting_name : !firrtl.const.uint<4>
+    %invalid = firrtl.invalidvalue : !firrtl.const.uint<4>
+    firrtl.strictconnect %y, %invalid : !firrtl.const.uint<4>
+  }
+  // CHECK-LABEL: module @Issue6819
+  firrtl.module @Issue6819() {
+    // CHECK: firrtl.connect %x, %{{[^ ]*}} : !firrtl.uint, !firrtl.const.uint<4>
+    firrtl.instance bar interesting_name @Bar()
+    %x = firrtl.wire interesting_name : !firrtl.uint
+    %invalid_ui1 = firrtl.invalidvalue : !firrtl.uint
+    firrtl.connect %x, %invalid_ui1 : !firrtl.uint, !firrtl.uint
+  }
```

Comment:
Went to do this but it triggers the special wire2node behavior added for GCView pipeline assumption reasons that I think makes it less obvious what the test is about (what happened to the wire? why? Is that relevant to whats being tested here?) and couples the test to that behavior unnecessarily.

Once found myself crafting a comment explaining why it becomes a node (and possibly what to expect in practice/otherwise) seems like maybe keeping it as-is is better as it's less surprising and doesn't need explanation.

I'll drop the other first "invalidate" to trim it a bit as you suggest.